src/extract/extract-regexes.pl

```diff
@@ -64,10 +65,29 @@
 }
 
 # Invoke the appropriate extractor.
+# If HTML file, extract the js part to create a new js file and pipeline it into the js extractor
+
 my $extractor = $language2extractor{$language};
 if ($extractor and -x $extractor) {
   print STDERR "$extractor '$json->{file}'\n";
+  my $original_path = $json->{file};
+
+  if ($language eq "html") {
```

Comment:
As discussed: I suggest you try an implementation for the html-extractor that recursively calls the meta-program to request JS extraction, and then parses its results.